manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -176,6 +176,17 @@ func (tr *TaskReaper) tick() {
 
 	defer func() {
 		tr.cleanup = nil
+
+		// Clean out the dirty set at the end of this tick iteration.
+		// When tick() finishes, the tasks in the slot were either cleaned up,
+		// or it was skipped because it didn't meet the criteria for cleaning.
+		// Either way, we can discard the dirty set because future events on that slot
+		// will cause the task to be readded to the dirty set at that point.
```

Comment:
I don't see anything that does it. The `EventUpdateTask` handler doesn't do anything with completed tasks unless they are also flagged for immediate removal.